src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -12,204 +12,191 @@
-                        promise.TrySetException(
-                            new Exception("Failed to write journal row batch",
-                                f.Cause));
-                        break;
-                    case QueueOfferResult.Dropped _:
-                        promise.TrySetException(new Exception(
-                            $"Failed to enqueue journal row batch write, the queue buffer was full ({_journalConfig.DaoConfig.BufferSize} elements)"));
-                        break;
-                    case QueueOfferResult.QueueClosed _:
-                        promise.TrySetException(new Exception(
-                            "Failed to enqueue journal row batch write, the queue was closed."));
-                        break;
-                }
-
-                return await promise.Task;
+                case QueueOfferResult.Enqueued _:
+                    break;
+                
+                case QueueOfferResult.Failure f:
+                    promise.TrySetException(new Exception("Failed to write journal row batch", f.Cause));
+                    break;
+                
+                case QueueOfferResult.Dropped _:
+                    promise.TrySetException(new Exception(
+                        $"Failed to enqueue journal row batch write, the queue buffer was full ({JournalConfig.DaoConfig.BufferSize} elements)"));
+                    break;
+                
+                case QueueOfferResult.QueueClosed _:
+                    promise.TrySetException(new Exception(
+                        "Failed to enqueue journal row batch write, the queue was closed."));
+                    break;
             }
+
+            await promise.Task;
         }
 
         private async Task WriteJournalRows(Seq<JournalRow> xs)
         {
+            //hot path:
+            //If we only have one row, penalty for BulkCopy
+            //Isn't worth it due to insert caching/transaction/etc.
+            switch (xs.Count)
             {
-                //hot path:
-                //If we only have one row, penalty for BulkCopy
-                //Isn't worth it due to insert caching/transaction/etc.
-                var count = xs.Count;
-                if (count > 1)
+                case 0:
+                    break;
+                case 1:
+                    await InsertSingle(xs);
+                    break;
+                default:
                     await InsertMultiple(xs);
-                else if (count == 1) await InsertSingle(xs);
+                    break;
             }
         }
 
         private async Task InsertSingle(Seq<JournalRow> xs)
         {
             //If we are writing a single row,
             //we don't need to worry about transactions.
-            using (var db = _connectionFactory.GetConnection())
-            {
-                await db.InsertAsync(xs.Head);
-            }
+            await using var db = ConnectionFactory.GetConnection();
```

Comment:
NIT/Don Quixote opinion:

I'm not a fain of Async dispose on Connections.

In the case of Connections, as long as you have a healthy connection pool you should normally not experience blocking on dispose and the task transitions aren't worth it from observations.